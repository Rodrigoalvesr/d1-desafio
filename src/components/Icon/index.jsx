import React from 'react';
import PropTypes from 'prop-types';
import icons from './list';
import { makeStyles } from '@material-ui/core/styles';
import classnames from 'classnames';

const useStyles = makeStyles((theme) => ({
  root: {
    fill: ({ color }) => color,
  },
}));

function Icon({ name, color, className, ...rest }) {
  const classes = useStyles({ color });
  const CustomIcon = icons[name];

  return (
    <CustomIcon className={classnames(classes.root, className)} {...rest} />
  );
}

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string,
};

export default Icon;
