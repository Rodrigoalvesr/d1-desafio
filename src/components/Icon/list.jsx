import { ReactComponent as ChartIcon } from '@/assets/icons/chart-pie.svg';
import { ReactComponent as RocketIcon } from '@/assets/icons/rocket.svg';
import { ReactComponent as UserFriendsIcon } from '@/assets/icons/user-friends.svg';
import { ReactComponent as CloudIcon } from '@/assets/icons/ccm-cloud.svg';
import { ReactComponent as ExternalLinkIcon } from '@/assets/icons/external-link-square-alt.svg';
import { ReactComponent as GemSolidIcon } from '@/assets/icons/gem-solid.svg';
import { ReactComponent as ToolsIcon } from '@/assets/icons/tools.svg';
import { ReactComponent as ExchangeIcon } from '@/assets/icons/exchange-alt.svg';
import { ReactComponent as LogoutIcon } from '@/assets/icons/sign-out-alt.svg';
import { ReactComponent as TableIcon } from '@/assets/icons/table.svg';
import { ReactComponent as SentIcon } from '@/assets/icons/paper-plane.svg';
import { ReactComponent as ActivatedIcon } from '@/assets/icons/play-circle.svg';
import { ReactComponent as ConfigurationIcon } from '@/assets/icons/pen.svg';
import { ReactComponent as IdleIcon } from '@/assets/icons/bed.svg';
import { ReactComponent as CompletedIcon } from '@/assets/icons/check.svg';

const icons = {
  chart: ChartIcon,
  rocket: RocketIcon,
  users: UserFriendsIcon,
  cloud: CloudIcon,
  link: ExternalLinkIcon,
  gem: GemSolidIcon,
  tools: ToolsIcon,
  exchange: ExchangeIcon,
  logout: LogoutIcon,
  table: TableIcon,
  sent: SentIcon,
  activated: ActivatedIcon,
  configuration: ConfigurationIcon,
  idle: IdleIcon,
  completed: CompletedIcon,
};

export default icons;
