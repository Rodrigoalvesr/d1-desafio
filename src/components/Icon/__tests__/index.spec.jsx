import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../index';

describe('Icon', () => {
  it('Renders correctly', () => {
    const component = renderer.create(<Icon name="chart" />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
