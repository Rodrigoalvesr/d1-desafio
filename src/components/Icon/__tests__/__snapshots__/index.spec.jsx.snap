// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Renders correctly 1`] = `
<svg
  className="makeStyles-root-1 makeStyles-root-2"
>
  chart-pie.svg
</svg>
`;
